### DISCUSSION TITLE
I cannot configure MQTT in Tasmota Version 12.0.2. I usually get it in version 8.5.1

### DISCUSSION THREAD

**Discussion by Grupo-Atomos** [arendst/Tasmota#15926] [Q&A] [answered]
### PROBLEM DESCRIPTION
I cannot configure MQTT in Tasmota Version 12.0.2. I usually get it in version 8.5.1

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ x ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ x ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ x ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ x ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ x ] Device used (e.g., Sonoff Basic): _____
- [ x ] Tasmota binary firmware version number used: 12.0.2
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: ESP8266
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
12:35:30.075 CMD: Backlog Template
12:35:30.142 RSL: RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}

12:46:31.565 CMD: Module; GPIO 255
12:46:31.569 RSL: RESULT = {"Module":{"1":"Sonoff Basic"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
12:47:22.989 CMD: Backlog Rule1
12:47:23.057 RSL: RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

12:47:56.326 CMD: Backlog Rule2
12:47:56.395 RSL: RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

12:49:09.376 CMD: Backlog Rule3
12:49:09.434 RSL: RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
12:49:45.507 CMD: Status 0
12:49:45.513 RSL: STATUS = {"Status":{"Module":1,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"afpesp/jardim","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
12:49:45.520 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"External System","Uptime":"0T00:15:19","StartupUTC":"2022-07-05T11:34:26","Sleep":50,"CfgHolder":4617,"BootCount":7,"BCResetTime":"2022-07-04T18:09:05","SaveCount":27,"SaveAddress":"F9000"}}
12:49:45.526 RSL: STATUS2 = {"StatusFWR":{"Version":"12.0.2(tasmota)","BuildDateTime":"2022-06-20T12:45:12","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"378/699"}}
12:49:45.531 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Atomos",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A190000000000","00000080","00006000","00004000","00000000"]}}
12:49:45.541 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":620,"Free":380,"Heap":27,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"1640E0","FlashFrequency":40,"FlashMode":2,"Features":["00000416","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","14000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
12:49:45.548 RSL: STATUS5 = {"StatusNET":{"Hostname":"afpesp-jardim-2015","IPAddress":"192.168.1.68","Gateway":"192.168.1.254","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.254","DNSServer2":"0.0.0.0","Mac":"5C:CF:7F:8B:C7:DF","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
12:49:45.551 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"broker.mqttdashboard.com","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_8BC7DF","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
12:49:45.558 RSL: STATUS7 = {"StatusTIM":{"UTC":"2022-07-05T11:49:45","Local":"2022-07-05T12:49:45","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"04:53","Sunset":"20:55"}}
12:49:45.563 RSL: STATUS10 = {"StatusSNS":{"Time":"2022-07-05T12:49:45"}}
12:49:45.568 RSL: STATUS11 = {"StatusSTS":{"Time":"2022-07-05T12:49:45","Uptime":"0T00:15:19","UptimeSec":919,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"ON","Wifi":{"AP":1,"SSId":"Atomos","BSSId":"04:25:E0:9B:82:A9","Channel":4,"Mode":"11n","RSSI":100,"Signal":-43,"LinkCount":1,"Downtime":"0T00:00:06"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
12:51:36.001 CMD: weblog 4
12:51:36.005 RSL: RESULT = {"WebLog":4}
12:51:36.835 CFG: Salvo na flash em F8, Contagem 28, Bytes 4096
12:51:48.282 MQT: Ligando...
12:51:48.516 MQT: A ligação falhou ao broker.mqttdashboard.com:1883, rc -2. Tentativa em 120 s
12:51:53.004 WIF: Verificando ligação...
12:52:13.029 WIF: Verificando ligação...
12:52:28.650 WIF: Sending Gratuitous ARP
12:52:33.034 WIF: Verificando ligação...
12:52:53.005 WIF: Verificando ligação...
12:53:13.035 WIF: Verificando ligação...
12:53:28.658 WIF: Sending Gratuitous ARP
12:53:33.033 WIF: Verificando ligação...

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
The wrong answer from version 12.0.2:

12:51:36.001 CMD: weblog 4
12:51:36.005 RSL: RESULT = {"WebLog":4}
12:51:36.835 CFG: Salvo na flash em F8, Contagem 28, Bytes 4096
12:51:48.282 MQT: Ligando...
12:51:48.516 MQT: A ligação falhou ao broker.mqttdashboard.com:1883, rc -2. Tentativa em 120 s
12:51:53.004 WIF: Verificando ligação...
12:52:13.029 WIF: Verificando ligação...
12:52:28.650 WIF: Sending Gratuitous ARP
12:52:33.034 WIF: Verificando ligação...
12:52:53.005 WIF: Verificando ligação...
12:53:13.035 WIF: Verificando ligação...
12:53:28.658 WIF: Sending Gratuitous ARP
12:53:33.033 WIF: Verificando ligação...
12:53:49.290 MQT: Ligando...
12:53:49.523 MQT: A ligação falhou ao broker.mqttdashboard.com:1883, rc -2. Tentativa em 120 s
12:53:53.012 WIF: Verificando ligação...
12:54:13.005 WIF: Verificando ligação...
12:54:28.677 WIF: Sending Gratuitous ARP
12:54:33.112 WIF: Verificando ligação...
12:54:33.670 RSL: STATE = {"Time":"2022-07-05T12:54:33","Uptime":"0T00:20:07","UptimeSec":1207,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"ON","Wifi":{"AP":1,"SSId":"Atomos","BSSId":"04:25:E0:9B:82:A9","Channel":4,"Mode":"11n","RSSI":100,"Signal":-43,"LinkCount":1,"Downtime":"0T00:00:06"}}
12:54:53.009 WIF: Verificando ligação...
12:55:13.018 WIF: Verificando ligação...
12:55:28.689 WIF: Sending Gratuitous ARP
12:55:33.008 WIF: Verificando ligação...
12:55:50.262 MQT: Ligando...
12:55:50.501 MQT: A ligação falhou ao broker.mqttdashboard.com:1883, rc -2. Tentativa em 120 s
12:55:53.008 WIF: Verificando ligação...

### EXPECTED BEHAVIOUR
After recording the firmware version 8.5.1, I put the address of the MQTT server (broker.mqttdashboard.com), then the topic "afpesp/jardim".
Saved the settings and everything works normally

The correct answer from version 8.5.1:
_A clear and concise description of what you expected to happen._
18:35:27 MQT: tele/afpesp/garden/LWT = Online (Keep)
18:35:27 MQT: cmnd/afpesp/garden/POWER =
18:35:27 MQT: tele/afpesp/jardim/INFO1 = {"Module":"Sonoff Basic","Version":"8.5.1(tasmota)","FallbackTopic":"cmnd/DVES_881AE5_fb/"," GroupTopic":"cmnd/tasmotas/"}
18:35:27 MQT: tele/afpesp/jardim/INFO2 = {"WebServerMode":"Admin","Hostname":"afpesp/jardim-6885","IPAddress":"192.168.1.81"}
18:35:27 MQT: tele/afpesp/jardim/INFO3 = {"RestartReason":"External System"}
18:35:27 MQT: stat/afpesp/jardim/RESULT = {"POWER":"OFF"}
18:35:27 MQT: stat/afpesp/garden/POWER = OFF

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._
With Tasmota 8.5.1:
![Tasmota 1](https://user-images.githubusercontent.com/13262479/177194617-0d8537d8-62ff-40a9-9513-3e5c3988bd9b.png)
![Tasmota2](https://user-images.githubusercontent.com/13262479/177195003-bb8991fe-0186-419d-b79e-65ab6a7d4fae.png)
![Tasmota 3](https://user-images.githubusercontent.com/13262479/177195303-58bd0276-2eab-44b9-b7fb-c65bb19f01c1.png)

With Tasmota 12.0.2:

![Tasmota 4](https://user-images.githubusercontent.com/13262479/177197379-90a5974f-8e30-49f9-bdeb-f2023926e2c5.png)
![Tasmota 5](https://user-images.githubusercontent.com/13262479/177197401-5245348d-7892-47f6-8988-b315dacc18a0.png)
![Tasmota 6](https://user-images.githubusercontent.com/13262479/177197411-6053d85c-541f-4a1e-bd93-f149c1b09779.png)

Visible differences of the two versions:

![Compara](https://user-images.githubusercontent.com/13262479/177201694-986ee89e-8710-43d2-a19e-a030316d037e.png)

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

Note that regardless of any other actions I may take, the MQTT settings are the same.
On Tasmota 8.5.1 everything works perfectly, on 12.0.2 it doesn't.
A while ago I opened a similar question with other versions here, but with no success.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by s-hadinger** [comment]
I just tried with 12.0.2 and it works fine.

**Reply by Grupo-Atomos** [reply]
@s-hadinger , Very strange, equipment not working in some places, and this same equipment in a previous version, works in this place.

**Comment by barbudor** [comment]
@Grupo-Atomos You haven't provided the logs with v12.0.2
Please edit your post and provide all the required information

**Reply by Grupo-Atomos** [reply]
@barbudor , I updated the logs

**Comment by barbudor** [comment]
As expected, worked immediately:
The only thing I set was the MQTT host and it connected seamlessly
No problems with v12.0.2
```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at FA, Count 10
00:00:00.055 QPC: Count 1
00:00:00.065 Project tasmota - Tasmota Version 12.0.2(tasmota)-2_7_4_9(2022-06-20T12:41:03)
00:00:00.505 WIF: Connecting to AP1 *********Channel 1 BSSId **:**:**:**:**:** in mode 11n as tasmota-2AD017-4119...
00:00:03.758 WIF: Connected
00:00:04.011 HTP: Web server active on tasmota-2AD017-4119 with IP address 192.168.168.52
20:28:12.028 MQT: Attempting connection...
20:28:12.101 MQT: Connected
20:28:12.104 MQT: tele/tasmota_2AD017/LWT = Online (retained)
20:28:12.106 MQT: cmnd/tasmota_2AD017/POWER = 
20:28:12.113 MQT: tele/tasmota_2AD017/INFO1 = {"Info1":{"Module":"Sonoff Basic","Version":"12.0.2(tasmota)","FallbackTopic":"cmnd/DVES_2AD017_fb/","GroupTopic":"cmnd/tasmotas/"}}
20:28:12.121 MQT: tele/tasmota_2AD017/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-2AD017-4119","IPAddress":"192.168.168.52"}}
20:28:12.133 MQT: tele/tasmota_2AD017/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":5}}
20:28:12.143 MQT: stat/tasmota_2AD017/RESULT = {"POWER":"OFF"}
20:28:12.148 MQT: stat/tasmota_2AD017/POWER = OFF
20:28:13.261 MQT: stat/tasmota_2AD017/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:00:08","StartupUTC":"2022-07-04T19:28:05","Sleep":50,"CfgHolder":4617,"BootCount":4,"BCResetTime":"2022-07-04T20:26:20","SaveCount":10,"SaveAddress":"FA000"}}
20:28:13.361 MQT: stat/tasmota_2AD017/STATUS11 = {"StatusSTS":{"Time":"2022-07-04T20:28:13","Uptime":"0T00:00:08","UptimeSec":8,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":21,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"BNETWLAN","BSSId":"88:C3:97:F2:25:EC","Channel":1,"Mode":"11n","RSSI":100,"Signal":-40,"LinkCount":1,"Downtime":"0T00:00:05"}}}
20:28:13.412 MQT: stat/tasmota_2AD017/RESULT = {"Time":"2022-07-04T20:28:13","Uptime":"0T00:00:08","UptimeSec":8,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":23,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"BNETWLAN","BSSId":"88:C3:97:F2:25:EC","Channel":1,"Mode":"11n","RSSI":100,"Signal":-39,"LinkCount":1,"Downtime":"0T00:00:05"}}
20:28:13.476 QPC: Reset
20:28:15.389 CMD: status 6
20:28:15.395 MQT: stat/tasmota_2AD017/STATUS6 = {"StatusMQT":{"MqttHost":"broker.mqttdashboard.com","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_2AD017","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
20:28:16.441 MQT: tele/tasmota_2AD017/STATE = {"Time":"2022-07-04T20:28:16","Uptime":"0T00:00:11","UptimeSec":11,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"******","BSSId":"**:**:**:**:**:**","Channel":1,"Mode":"11n","RSSI":100,"Signal":-38,"LinkCount":1,"Downtime":"0T00:00:05"}}
```

**Reply by Grupo-Atomos** [reply]
@Grupo-Atomos , I checked the TimeZone, I changed it to Time Zone 3, but it still doesn't work here.

**Reply by Grupo-Atomos** [reply]
@barbudor, this is the logs for another ESP8266, with the Tasmota version 8.5.1 connected at the same time in my network:

00:00:00 CFG: Lido da flash em F9, Contagem 11
00:00:00 QPC: Count 1
00:00:00 Projeto tasmota Tasmota Versão 8.5.1(tasmota)-2_7_4_1
00:00:00 WIF: Ligando ao PA1 Atomos Channel 4 BSSId 04:25:E0:9B:82:A9 em modo 11N como afpesp/jardim-6885...
00:00:01 WIF: Ligado
15:21:26 HTP: Servidor WEB ativo em afpesp/jardim-6885 com o endereço IP 192.168.1.81
15:21:27 MQT: Ligando...
15:21:27 MQT: Ligado
15:21:27 MQT: tele/afpesp/jardim/LWT = Online (Manter)
15:21:27 MQT: cmnd/afpesp/jardim/POWER = 
15:21:27 MQT: tele/afpesp/jardim/INFO1 = {"Module":"Sonoff Basic","Version":"8.5.1(tasmota)","FallbackTopic":"cmnd/DVES_881AE5_fb/","GroupTopic":"cmnd/tasmotas/"}
15:21:27 MQT: tele/afpesp/jardim/INFO2 = {"WebServerMode":"Admin","Hostname":"afpesp/jardim-6885","IPAddress":"192.168.1.81"}
15:21:27 MQT: tele/afpesp/jardim/INFO3 = {"RestartReason":"Software/System restart"}
15:21:27 MQT: stat/afpesp/jardim/RESULT = {"POWER":"OFF"}
15:21:27 MQT: stat/afpesp/jardim/POWER = OFF
15:21:28 MQT: stat/afpesp/jardim/RESULT = {"POWER":"OFF"}
15:21:28 MQT: stat/afpesp/jardim/POWER = OFF
15:21:30 QPC: Reset
15:21:31 MQT: tele/afpesp/jardim/STATE = {"Time":"2022-07-05T15:21:31","Uptime":"0T00:00:09","UptimeSec":9,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Atomos","BSSId":"04:25:E0:9B:82:A9","Channel":4,"RSSI":100,"Signal":-33,"LinkCount":1,"Downtime":"0T00:00:03"}}

**Comment by s-hadinger** [comment]
To be fair, I tested with ESP32 not ESP8266. I will try later.

**Comment by barbudor** [comment, marked as answer]
@Grupo-Atomos it is a network failure to connect
if your internet connection is slow you may need to increase `MqttWifiTimeout`
The value was originally hard-coded to 5000 ms before v10 or so
The value was then reduced to 200ms which is fine for Lan but may be too short for slow connections
Try `MqttWifiTimeout 2000`

**Reply by Grupo-Atomos** [reply]
Thank you very much to everyone in this chat, but I couldn't help but thank Master @barbudor  in particular for his dedication, patience and generosity.
Thank you my friend!

Here are the logs after the mqttwifitimeout 2000 command:

00:00:00.001 HDW: ESP8266EX
00:00:00.049 CFG: Lido da flash em FA, Contagem 10
00:00:00.054 QPC: Count 1
00:00:00.063 Projeto tasmota - Tasmota Versão 12.0.2(tasmota)-2_7_4_9(2022-06-20T12:45:12)
00:00:00.551 WIF: Ligando ao PA1 Atomos Channel 7 BSSId 04:25:E0:9B:82:A9 em modo 11n como afpesp-jardim-2015...
00:00:01.801 WIF: Ligado
00:00:02.003 HTP: Servidor WEB ativo em afpesp-jardim-2015 com o endereço IP 192.168.1.68
22:45:09.028 MQT: Ligando...
22:45:09.269 MQT: A ligação falhou ao broker.mqttdashboard.com:1883, rc -2. Tentativa em 10 s
22:45:12.453 QPC: Reset
22:45:20.092 MQT: Ligando...
22:45:20.324 MQT: A ligação falhou ao broker.mqttdashboard.com:1883, rc -2. Tentativa em 20 s
22:45:41.031 MQT: Ligando...
22:45:41.264 MQT: A ligação falhou ao broker.mqttdashboard.com:1883, rc -2. Tentativa em 30 s

22:45:54.056 CMD: mqttwifitimeout 2000
22:45:54.060 RSL: RESULT = {"MqttWifiTimeout":2000}
22:46:12.036 MQT: Ligando...
22:46:12.506 MQT: Ligado
22:46:12.509 MQT: tele/afpesp/jardim/LWT = Online (Manter)
22:46:12.511 MQT: cmnd/afpesp/jardim/POWER = 
22:46:12.518 MQT: tele/afpesp/jardim/INFO1 = {"Info1":{"Module":"Sonoff Basic","Version":"12.0.2(tasmota)","FallbackTopic":"cmnd/DVES_8BC7DF_fb/","GroupTopic":"cmnd/tasmotas/"}}
22:46:12.526 MQT: tele/afpesp/jardim/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"afpesp-jardim-2015","IPAddress":"192.168.1.68"}}
22:46:12.537 MQT: tele/afpesp/jardim/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":4}}
22:46:12.548 MQT: stat/afpesp/jardim/RESULT = {"POWER":"OFF"}
22:46:12.552 MQT: stat/afpesp/jardim/POWER = OFF
22:46:14.077 MQT: stat/afpesp/jardim/RESULT = {"Time":"2022-07-05T22:46:14","Uptime":"0T00:01:10","UptimeSec":70,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":29,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Atomos","BSSId":"04:25:E0:9B:82:A9","Channel":7,"Mode":"11n","RSSI":100,"Signal":-40,"LinkCount":1,"Downtime":"0T00:00:03"}}
22:46:16.391 MQT: tele/afpesp/jardim/STATE = {"Time":"2022-07-05T22:46:16","Uptime":"0T00:01:12","UptimeSec":72,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Atomos","BSSId":"04:25:E0:9B:82:A9","Channel":7,"Mode":"11n","RSSI":100,"Signal":-35,"LinkCount":1,"Downtime":"0T00:00:03"}}
22:47:06.138 MQT: stat/afpesp/jardim/RESULT = {"POWER":"OFF"}
22:47:06.140 MQT: stat/afpesp/jardim/POWER = OFF
22:47:31.712 MQT: stat/afpesp/jardim/RESULT = {"POWER":"ON"}
22:47:31.715 MQT: stat/afpesp/jardim/POWER = ON

Now tell me, if it's not asking too much: at what point did you realize that the problem was due to the MQTT Wifi Connection Timeout?

**Reply by barbudor** [reply]
> Now tell me, if it's not asking too much: at what point did you realize that the problem was due to the MQTT Wifi Connection Timeout?

A few clues suggested me that it could be the problem:
- it is working with version prior to 9.5.0
- the logs shows the connection error (-2) and your parameters seems ok (not the same problems as in our previous [discussion](https://github.com/arendst/Tasmota/discussions/13180) )
- I think I remember you are in Brazil. For me broker.mqttdashboard.com appears to be in Germany, so close to me (France). A typicall Internet round-trip-time across the world is around 200-250ms. So may not be a problem with the local network but the overall distance.

For curiosity, could you please check with [iplocation.net](https://iplocation.net/ip-lookup) where is located broker.mqttdashboard.com when tested from your side ?

I would be interrested to know which is the smallest MqttWifiTimeout value that gives you a consistent connection.
I may suggest that we raise the default value to something higher as between the old value of 5000ms and the current of 200ms there might be better compromise.

**Reply by arendst** [reply]
Great find!!

If we can distinguish between local and remote mqtt servers we can set the timeout automatically to 200 local and 1000 remote.

**Reply by barbudor** [reply]
A broker which IP address is not in a private network range such as 10.0.0.0/24, 172.16.0.0/21, 192.168.0.0/16 would be in the cloud

But I was mainly suggesting to pump up the default value to 400ms or 500ms as a compromise
I don't expect an round-trip-time on the Internet to be bigger than 250ms

**Reply by barbudor** [reply]
I'm not sure if the timeout stops on the 1st packet received from the remote broker or until the full transaction is completed.
If full transaction to open the socket is needed, it may be longuer, especially if TLS is involved
🤔

**Reply by barbudor** [reply]
A strategy could be that when we increase the time between attemps, we also increase the timeout (RAM, not saving)
And when/if the connection finally occurs, we display the value so users would be aware that a longer than default value is required

**Reply by sfromis** [reply]
A strategy of increasing timeout when retrying should probably still have a low max, due to the reason for the default timeout being pretty low, and most retry situation would just be due to "normal" network trouble precluding access.

**Reply by Grupo-Atomos** [reply]
> > Agora me diga, se não for pedir muito: em que momento você percebeu que o problema era devido ao tempo limite da conexão Wifi MQTT?
> 
> Algumas pistas me sugeriram que poderia ser o problema:
> 
> * está trabalhando com a versão anterior a 9.5.0
> * os logs mostram o erro de conexão (-2) e seus parâmetros parecem ok (não os mesmos problemas que em nossa [discussão](https://github.com/arendst/Tasmota/discussions/13180) anterior )
> * Acho que me lembro que você está no Brasil. Para mim, broker.mqttdashboard.com parece estar na Alemanha, tão perto de mim (França). Um tempo de ida e volta típico da Internet em todo o mundo é de cerca de 200-250ms. Portanto, pode não ser um problema com a rede local e sim com a distância geral.
> 
> Por curiosidade, você poderia verificar com [iplocation.net](https://iplocation.net/ip-lookup) onde está localizado broker.mqttdashboard.com quando testado do seu lado?
> 
> Eu estaria interessado em saber qual é o menor valor MqttWifiTimeout que fornece uma conexão consistente. Posso sugerir que aumentemos o valor padrão para algo mais alto, pois entre o valor antigo de 5000ms e o atual de 200ms pode haver um melhor compromisso.

@barbudor, Perfect!
Here's the geolocation:

01:
![Folha 01](https://user-images.githubusercontent.com/13262479/177879256-4574afcb-7a02-45ca-9a93-04842bc2e46b.png)

02:
![Folha 02](https://user-images.githubusercontent.com/13262479/177879324-70fef739-40ef-48ce-a28d-a7743cd5b63c.png)

Thanks!

**Reply by barbudor** [reply]
Thanks, that confirm that you and the broker are quite far away from each other.
